### ISSUE TITLE
SAM Model Predict Bug

### ISSUE THREAD

**Issue by PtttCode** [heartexlabs/label-studio-ml-backend#278]
I had followed https://github.com/heartexlabs/label-studio-ml-backend/tree/master/label_studio_ml/examples/segment_anything_model to deploy **SAM model**
But there are some bugs of **SAM** predictions
Exp:
1. Here is a 2048*1024 image performance:
![图片](https://github.com/heartexlabs/label-studio-ml-backend/assets/36265886/832fc00a-20d8-4aef-b220-6837b72185d4)

2. Here is a 640*426 image performance:
![图片](https://github.com/heartexlabs/label-studio-ml-backend/assets/36265886/9436ed4c-6056-48c0-a2b6-ece2487f25e6)

3. Here is a 640*480 image performance:
![图片](https://github.com/heartexlabs/label-studio-ml-backend/assets/36265886/79cf8f48-6d1e-49dd-a196-92a64bdc1691)

There are absolutely bugs on the process after prediction, and I think the key of question is the image padding

Here are the original images:
![88a3f9da-berlin_000005_000019_leftImg8bit](https://github.com/heartexlabs/label-studio-ml-backend/assets/36265886/bf96561a-7ebc-4aab-8cb4-c362eeb710ba)
![d6d991d3-000000055150](https://github.com/heartexlabs/label-studio-ml-backend/assets/36265886/de566496-f4ef-4daa-9759-badeddd1e0ca)
![d01284fc-000000167486](https://github.com/heartexlabs/label-studio-ml-backend/assets/36265886/940c416b-a158-4cf8-a0d6-fc904900554f)

**Comment by jflessau** [comment]
Probably related to this issue with the ONNX model: https://github.com/anuragxel/salt/issues/2

Seems like masks are off for input images not matching the aspect ratio (2:3) set with 

```python
"orig_im_size": torch.tensor([1500, 2250], dtype=torch.float),
```

in [onnxconverter.py](https://github.com/heartexlabs/label-studio-ml-backend/blob/b2c31d34c0414befde3d0d067588617e062ef443/label_studio_ml/examples/segment_anything_model/onnxconverter.py#L32)

Your example image, with a size of 640x426, happens to have the correct aspect ratio.

**Comment by shondle** [comment]
Hi, this issue was addressed in the pull request mentioned. @jflessau thank you so much for pointing me in the right direction. This new pull request (if using the AdvancedSAM backend instead) should have no issues with masks offsetting from images -> this is an ONNX issue [described in the README](https://github.com/shondle/label-studio-ml-backend/tree/mobile_sam/label_studio_ml/examples/segment_anything_model#notes-for-onnx) of this pull request.
